openupgrade_scripts/scripts/payment/17.0.2.0/pre-migration.py

```diff
@@ -0,0 +1,189 @@
+        "payment.payment_icon_cc_jcb",
+        "payment.payment_method_jcb",
+    ),
+    (
+        "payment.payment_icon_cc_maestro",
+        "payment.payment_method_maestro",
+    ),
+    (
+        "payment.payment_icon_cc_cirrus",
+        "payment.payment_method_cirrus",
+    ),
+    (
+        "payment.payment_icon_cc_unionpay",
+        "payment.payment_method_unionpay",
+    ),
+    (
+        "payment.payment_icon_cc_bancontact",
+        "payment.payment_method_bancontact",
+    ),
+    # This one has been removed with https://github.com/odoo/odoo/pull/140907
+    # (
+    #     "payment.payment_icon_cc_western_union",
+    #     "payment.payment_method_western_union",
+    # ),
+    (
+        "payment.payment_icon_sepa",
+        "payment.payment_method_sepa_direct_debit",
+    ),
+    (
+        "payment.payment_icon_cc_ideal",
+        "payment.payment_method_ideal",
+    ),
+    # This one has been removed with https://github.com/odoo/odoo/pull/120446
+    #
+    # (
+    #     "payment.payment_icon_cc_webmoney",
+    #     "payment.payment_method_webmoney",
+    # ),
+    (
+        "payment.payment_icon_cc_giropay",
+        "payment.payment_method_giropay",
+    ),
+    (
+        "payment.payment_icon_cc_eps",
+        "payment.payment_method_eps",
+    ),
+    (
+        "payment.payment_icon_cc_p24",
+        "payment.payment_method_p24",
+    ),
+    (
+        "payment.payment_icon_cc_codensa_easy_credit",
+        "payment.payment_method_codensa",
+    ),
+    (
+        "payment.payment_icon_kbc",
+        "payment.payment_method_kbc_cbc",
+    ),
+    (
+        "payment.payment_icon_mpesa",
+        "payment.payment_method_mpesa",
+    ),
+    # This one has been removed with https://github.com/odoo/odoo/pull/120446
+    # (
+    #     "payment.payment_icon_airtel_money",
+    #     "payment.payment_method_airtel_money",
+    # ),
+    (
+        "payment.payment_icon_mtn_mobile_money",
+        "payment.payment_method_mobile_money",
+    ),
+    (
+        "payment.payment_icon_barter_by_flutterwave",
+        "payment.payment_method_ussd",
+    ),
+    # This one has been removed with https://github.com/odoo/odoo/pull/120446
+    # (
+    #     "payment.payment_icon_sadad",
+    #     "payment.payment_method_sadad",
+    # ),
+    (
+        "payment.payment_icon_mada",
+        "payment.payment_method_mada",
+    ),
+    # These ones have been removed with https://github.com/odoo/odoo/pull/120446
+    # (
+    #     "payment.payment_icon_bbva_bancomer",
+    #     "payment.payment_method_bbva_bancomer",
+    # ),
+    # (
+    #     "payment.payment_icon_citibanamex",
+    #     "payment.payment_method_citibanamex",
+    # ),
+]
+
+_model_renames = [
+    ("payment.icon", "payment.method"),
+]
+
+_table_renames = [
+    ("payment_icon", "payment_method"),
+]
+
+_field_renames = [
+    (
+        "payment.provider",
+        "payment_provider",
+        "payment_icon_ids",
+        "payment_method_ids",
+    ),
```

Comment:
You are right of course, however since link between provider and methods completely changed in v17, I thought that maybe it was better to let ORM recreate the full table.
Now I see that you may have extra payment provider (not provided by Odoo) that would need to keep their corresponding methods, so it makes sense to migrate those as you did in https://github.com/OCA/OpenUpgrade/pull/4692/commits/4ec754cc2bc785841ec95388a3cf4df403b1901c